Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +181,293 @@ void SkylanderPortalWindow::CreateMainWindow()
-    hbox_skylander->addWidget(clear_btn);
-    hbox_skylander->addWidget(create_btn);
-    hbox_skylander->addWidget(load_btn);
 
     vbox_group->addLayout(hbox_skylander);
   }
+  m_slot_radios[0]->setChecked(true);
 
   m_group_skylanders->setLayout(vbox_group);
+  m_group_skylanders->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Preferred);
   scroll_area->setWidget(m_group_skylanders);
   scroll_area->setWidgetResizable(true);
+  scroll_area->setFrameStyle(QFrame::NoFrame);
   m_group_skylanders->setVisible(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
-  main_layout->addWidget(scroll_area);
-  setLayout(main_layout);
+  portal_slots_layout->addWidget(scroll_area);
+  portal_slots_group->setLayout(portal_slots_layout);
+  slot_layout->addWidget(portal_slots_group);
 
-  UpdateEdits();
+  return slot_layout;
 }
 
-void SkylanderPortalWindow::OnEmulationStateChanged(Core::State state)
+QVBoxLayout* SkylanderPortalWindow::CreateFinderLayout()
 {
-  const bool running = state != Core::State::Uninitialized;
+  auto* main_layout = new QVBoxLayout();
+
+  // WIDGET: Skylander List
+  m_skylander_list = new QListWidget;
+  connect(m_skylander_list, &QListWidget::itemDoubleClicked, this,
+          &SkylanderPortalWindow::LoadSelected);
+
+  auto* collection_layout = new QHBoxLayout();
+
+  // WIDGET: Skylander Collection Path
+  collection_layout->addWidget(new QLabel(tr("Skylander Collection Path:")));
+  m_path_edit = new QLineEdit;
+  collection_layout->addWidget(m_path_edit);
+  m_path_select = new QPushButton(tr("Choose"));
+  connect(m_path_edit, &QLineEdit::editingFinished, this,
+          &SkylanderPortalWindow::OnCollectionPathChanged);
+  connect(m_path_select, &QAbstractButton::clicked, this,
+          &SkylanderPortalWindow::SelectCollectionPath);
+  collection_layout->addWidget(m_path_select);
+
+  main_layout->addLayout(collection_layout);
+
+  auto* finder_layout = new QVBoxLayout();
+
+  // WIDGET: Search Bar
+  auto* search_bar_layout = new QHBoxLayout;
+  m_sky_search = new QLineEdit;
+  m_sky_search->setClearButtonEnabled(true);
+  connect(m_sky_search, &QLineEdit::textChanged, this, &SkylanderPortalWindow::RefreshList);
+  search_bar_layout->addWidget(new QLabel(tr("Search:")));
+  search_bar_layout->addWidget(m_sky_search);
+  finder_layout->addLayout(search_bar_layout);
+
+  auto* finder_list_layout = new QHBoxLayout();
+
+  auto* search_filters_layout = new QVBoxLayout();
+
+  // WIDGET: Filter by Game
+  auto* search_game_group = new QGroupBox(tr("Game"));
+  auto* search_game_layout = new QVBoxLayout();
+  auto* search_checkbox_scroll_area = new QScrollArea();
+  search_checkbox_scroll_area->setContentsMargins(0, 0, 0, 0);
+  search_checkbox_scroll_area->setFrameStyle(QFrame::NoFrame);
+  auto* search_checkbox_group = new QFrame();
+  search_checkbox_group->setContentsMargins(0, 0, 0, 0);
+  auto* search_checkbox_layout = new QVBoxLayout();
+
+  for (int i = 0; i < 5; i++)
```

Comment:
When `m_game_filters` is an `std::array` you can instead do `for (size_t i = 0; i < m_game_filters.size(); ++i)`